src/main/groovy/net/researchgate/release/ReleasePlugin.groovy

```diff
@@ -28,12 +28,19 @@ import org.gradle.api.GradleException
 import org.gradle.api.Plugin
 import org.gradle.api.Project
 import org.gradle.api.Task
+import org.gradle.api.model.ObjectFactory
 import org.gradle.api.plugins.BasePlugin
+import org.gradle.api.provider.ProviderFactory
 import org.gradle.api.tasks.GradleBuild
 import org.gradle.api.tasks.TaskState
+import org.gradle.build.event.BuildEventsListenerRegistry
+import org.gradle.tooling.events.OperationCompletionListener
+import org.gradle.tooling.events.task.TaskFailureResult
+import org.gradle.util.GradleVersion
 
-class ReleasePlugin extends PluginHelper implements Plugin<Project> {
+import javax.inject.Inject
 
+abstract class ReleasePlugin extends PluginHelper implements Plugin<Project> {
```

Comment:
What is the reason for this change?
I see that you added
```
    @Inject
    abstract protected ObjectFactory getObjects();

    @Inject
    abstract protected ProviderFactory getProviders();
 ```
But I don't get the why.

Reply:
Because I use them at https://github.com/researchgate/gradle-release/blob/7ffe081f69545e66ad4625ee5ed68e7714c44b36/src/main/groovy/net/researchgate/release/ReleasePlugin.groovy#L174 and https://github.com/researchgate/gradle-release/blob/7ffe081f69545e66ad4625ee5ed68e7714c44b36/src/main/groovy/net/researchgate/release/ReleasePlugin.groovy#L176

Of course you can get those from `project` too, but that is bad / outdated practice and I didn't want to add to the existing ones but doing it right. :-)